src/macros.jl

```diff
@@ -35,6 +35,37 @@ function _add_kw_args(call, kw_args)
     end
 end
 
+"""
+    _add_positional_args(call, args)::Nothing
+
+Add the positional arguments `args` to the function call expression `call`, 
+escaping each argument expression. The elements of `args` should should be ones 
+that were extracted via [`Containers._extract_kw_args`](@ref) and had appropriate 
+arguments filtered out (e.g., the model argument). This is able to incorporate 
+additional positional arguments to `call`s that already have keyword arguments.
+
+## Examples 
+
+```jldoctest; setup = :(using JuMP)
+julia> call = :(f(1, a=2))
+:(f(1, a = 2))
+
+julia> JuMP._add_positional_args(call, [:(x)])
+
+julia> call
+:(f(1, $(Expr(:escape, :x)), a = 2))
+```
+"""
+function _add_positional_args(call, args)
```

Comment:
Can we force a single positional argument for now?
Multiple ones can be very hard to deal with.

Reply:
Wouldn't that be inconsistent with `@variable' which allows multiple extra arguments?

Reply:
Yes it would be inconsistent.
But I dont think that was a food decision.
I think we should remove that behaviour from the variable macro.
Note that there, many positional args are no actually positional because the are used in any order, almost as keywords.
We should allow a single user defined positional arg. Otherwise the combinatorics of methods that need to be implemented will lead to unnecessary errors.
If some interface really needs two, they could create an arg that is the combination of two.

Reply:
ref this https://github.com/jump-dev/JuMP.jl/pull/2421#issuecomment-774062945
and all the above comments.

Reply:
Moreover, allowing more than one later is non-breaking.
While constraining to one later is breaking.

Reply:
opened an issue: https://github.com/jump-dev/JuMP.jl/issues/2583

Reply:
That makes sense, I have updated everything accordingly. This is enforced in `_constraint_macro` and not `_add_positional_args` to provide better error messages and leave `_add_positional_args` as a general method.